Revit_Core_Engine/Convert/Lighting/ToRevit/Luminaire.cs

```diff
@@ -0,0 +1,84 @@
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Autodesk.Revit.DB;
+using Autodesk.Revit.DB.Structure;
+using BH.Engine.Adapters.Revit;
+using BH.oM.Adapters.Revit.Elements;
+using BH.oM.Adapters.Revit.Settings;
+using BH.oM.Base;
+using BH.oM.Base.Attributes;
+using BH.oM.Geometry;
+using BH.oM.Lighting.Elements;
+using BH.oM.Physical.Elements;
+using BH.oM.Spatial.SettingOut;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Convert
+    {
+        /***************************************************/
+        /****               Public Methods              ****/
+        /***************************************************/
+
+        [Description("Converts BH.oM.Lighting.Elements.Luminaire to a Revit LightingFixture.")]
+        [Input("luminaire", "BH.oM.Lighting.Elements.Luminaire to be converted.")]
+        [Input("document", "Revit document, in which the output of the convert will be created.")]
+        [Input("settings", "Revit adapter settings to be used while performing the convert.")]
+        [Input("refObjects", "Optional, a collection of objects already processed in the current adapter action, stored to avoid processing the same object more than once.")]
+        [Output("lightingFixture", "Revit LightingFixture element resulting from the BH.oM.Lighting.Elements.Luminaire.")]
+        public static FamilyInstance ToRevitFamilyInstance (this Luminaire luminaire, Document document, RevitSettings settings = null, Dictionary<Guid, List<int>> refObjects = null)
+        {
+            if (luminaire == null || document == null)
+                return null;
+
+            FamilyInstance familyInstance = refObjects.GetValue<FamilyInstance>(document, luminaire.BHoM_Guid);
+            if (familyInstance != null)
+                return familyInstance;
+
+            settings = settings.DefaultIfNull();
+
+            FamilySymbol familySymbol = (FamilySymbol)luminaire.ElementType(document, settings);
+            if (familySymbol == null)
+            {
+                Compute.ElementTypeNotFoundWarning(luminaire);
+                return null;
+            }
+
+            if (luminaire.Position != null)
+                return Create.FamilyInstance(document, familySymbol, (luminaire.Position).ToRevit(), luminaire.Orientation.ToRevit(), luminaire.HostElement(document, settings), settings);
+            else
+            {
+                BH.Engine.Base.Compute.RecordError($"A family could not be created based on the given luminaire because its location was invalid. BHoM_Guid: {luminaire.BHoM_Guid}");
```

Comment:
```suggestion
                BH.Engine.Base.Compute.RecordError($"An element could not be created based on the given luminaire because its location was invalid. BHoM_Guid: {luminaire.BHoM_Guid}");
```

Reply:
fixed